grapher/mapCharts/MapTooltip.tsx

```diff
@@ -125,93 +130,165 @@ export class MapTooltip extends React.Component<MapTooltipProps> {
+                    { value: Infinity, priority: 2 },
+                ],
+            },
+            xAxisConfig: {
+                hideAxis: false,
+                hideGridlines: true,
+                compactLabels: true,
+                // Always show up to the target time on the X axis,
+                // even if there is no data for it.
+                min: this.props.targetTime,
+                max: this.props.targetTime,
+                ticks: [
+                    // Show minimum and maximum
+                    { value: -Infinity, priority: 1 },
+                    { value: Infinity, priority: 1 },
+                ],
+            },
+        }
+    }
+
     render(): JSX.Element {
-        const { tooltipDatum, isEntityClickable, targetTime, manager } =
-            this.props
+        const { tooltipTarget, mapTable, datum, textColorScale } = this
+        const { isEntityClickable, targetTime, manager } = this.props
 
+        // Only LineChart and ScatterPlot allow `mapIsClickable`
         const clickToSelectMessage =
             manager.type === ChartTypeName.LineChart
                 ? "Click for change over time"
                 : "Click to select"
 
-        const { timeColumn } = this.inputTable
-        const { renderSparkBars, barColor, tooltipTarget } = this
-
+        // TODO simplify?
+        const { timeColumn } = mapTable
+        const preposition = getPreposition(timeColumn)
         const displayTime = !timeColumn.isMissing
             ? timeColumn.formatValue(targetTime)
             : targetTime
         const displayDatumTime =
-            timeColumn && tooltipDatum
-                ? timeColumn.formatValue(tooltipDatum.time)
-                : tooltipDatum?.time.toString() ?? ""
+            timeColumn && datum
+                ? timeColumn.formatValue(datum?.time)
+                : datum?.time.toString() ?? ""
+        const valueColor: string | undefined = darkenColorForHighContrastText(
+            textColorScale?.getColor(datum?.value) ?? "#333"
+        )
 
         return (
             <Tooltip
                 tooltipManager={this.props.manager}
                 key="mapTooltip"
                 x={tooltipTarget.x}
                 y={tooltipTarget.y}
-                style={{ textAlign: "center", padding: "8px" }}
+                style={{ textAlign: "center", padding: "0.3em" }}
                 offsetX={15}
                 offsetY={10}
                 offsetYDirection={"upward"}
             >
-                <h3
+                <div
                     style={{
-                        padding: "0.3em 0.3em",
+                        padding: "0.3em",
                         margin: 0,
                         fontWeight: "normal",
                         fontSize: "1em",
                     }}
                 >
                     {tooltipTarget.featureId ||
                         tooltipTarget.featureId.replace(/_/g, " ")}
```

Comment:
I know this is not something you touched, but this right here is just extremely weird? The part after `||` is always empty when it is reached?

Reply:
Yes! It's weird. I think `&&` was maybe intended. But even then, I can't see why we would replace underscores in the tooltip only? I also can't see how `_` would appear in the string, there don't seem to be underscores in `MapTopology.ts`. 

I will go ahead and remove the stuff after `||`, let me know if you have ideas.